crates/optimism/evm/src/l1.rs

```diff
@@ -246,6 +246,19 @@ pub trait RethL1BlockInfo {
     ) -> Result<U256, BlockExecutionError>;
 }
 
+fn determine_spec_id(
+    chain_spec: &impl OpHardforks,
+    timestamp: u64,
+    block_number: u64,
+) -> Result<OpSpecId, BlockExecutionError> {
+    let spec_id = revm_spec_by_timestamp_after_bedrock(chain_spec, timestamp);
+    if spec_id == OpSpecId::BEDROCK && !chain_spec.is_bedrock_active_at_block(block_number) {
```

Comment:
since we expect that bedrock is active, we can probably even get rid of this bedrock check as well

Reply:
agreed. that also makes this whole function redundant, let's remove it and just call `revm_spec_by_timestamp_after_bedrock` in the scope of the `L1BlockInfo` methods.